internal/testutils/tls_gen.go

```diff
@@ -72,9 +77,13 @@ func (c *CertBuilder) Build() (*ServerCerts, *ClientCerts, error) {
 	}
 
 	caCertPem := bytes.Buffer{}
-	err = pem.Encode(&caCertPem, &pem.Block{Type: "CERTIFICATE", Bytes: caCertBytes})
-	if err != nil {
-		return nil, nil, err
+	if !c.caInvalid {
+		err = pem.Encode(&caCertPem, &pem.Block{Type: "CERTIFICATE", Bytes: caCertBytes})
```

Comment:
Please do some investigations if there could be other possible headers for a CA bundle.

Reply:
According to this RFC, you can expect the following labels to be within the BEGIN header and END footer:

CERTIFICATE [[RFC5280]](https://www.rfc-editor.org/rfc/rfc5280)
X509 CRL [[RFC5280]](https://www.rfc-editor.org/rfc/rfc5280)
CERTIFICATE REQUEST [[RFC2986]](https://www.rfc-editor.org/rfc/rfc2986)
PKCS7 [[RFC2315]](https://www.rfc-editor.org/rfc/rfc2315)
CMS [[RFC5652]](https://www.rfc-editor.org/rfc/rfc5652)
PRIVATE KEY [[RFC5208]](https://www.rfc-editor.org/rfc/rfc5208) [[RFC5958]](https://www.rfc-editor.org/rfc/rfc5958)
ENCRYPTED PRIVATE KEY [[RFC5958]](https://www.rfc-editor.org/rfc/rfc5958)
ATTRIBUTE CERTIFICATE [[RFC5755]](https://www.rfc-editor.org/rfc/rfc5755)
PUBLIC KEY [[RFC5280]](https://www.rfc-editor.org/rfc/rfc5280)

> More or less, CAs are just normal certificates with some specific values, so from my perspective, we should be fine

> **_Source: https://stackoverflow.com/questions/5355046/where-is-the-pem-file-format-specified_**